CMakeLists.txt

```diff
@@ -45,15 +45,18 @@ include(CheckCSourceRuns)
 include(CheckCCompilerFlag)
 include(CMakeDependentOption)
 include(CMakePackageConfigHelpers)
+include(CTest)
 include(FeatureSummary)
 
 # We need to enable C++ before trying to check for coverage
-option(ZLIB_ENABLE_TESTS "Build test binaries" ON)
-cmake_dependent_option(ZLIBNG_ENABLE_TESTS "Test zlib-ng specific API" ON "ZLIB_ENABLE_TESTS" OFF)
-cmake_dependent_option(WITH_GTEST "Build gtest_zlib" ON "ZLIB_ENABLE_TESTS" OFF)
-cmake_dependent_option(WITH_FUZZERS "Build test/fuzz" OFF "ZLIB_ENABLE_TESTS" OFF)
-cmake_dependent_option(WITH_BENCHMARKS "Build test/benchmarks" OFF "ZLIB_ENABLE_TESTS" OFF)
-cmake_dependent_option(WITH_BENCHMARK_APPS "Build application benchmarks" OFF "WITH_BENCHMARKS" OFF)
+if (DEFINED ZLIB_ENABLE_TESTS AND NOT ZLIB_ENABLE_TESTS)
+    message(DEPRECATION "ZLIB_ENABLE_TESTS is deprecated. Please use BUILD_TESTING instead.")
```

Comment:
This line shouldn't depend on value of `ZLIB_ENABLE_TESTS`, only on that the variable is defined to something.

Reply:
Indeed. Fixed.

Reply:
"DEPRECATION" as first parameter in message() command is not supported until CMake 3.17

Reply:
The latest version says it was added in 3.17... Either they changed the meaning of the type, or there is error in either one of the document versions. It's something that needs to be verified...

Reply:
@mtl1979 Link? I can't find that in any of the most recent versions, or the git-master.
Git master does say "Added in version 3.15: Added the NOTICE, VERBOSE, DEBUG, and TRACE levels."
Nothing about the DEPRECATION one though.
Ref: https://cmake.org/cmake/help/git-master/command/message.html